internal/action/command.go

```diff
@@ -535,8 +536,10 @@ func SetGlobalOptionNative(option string, nativeValue interface{}) error {
 	}
 
 	// ...if it's not local continue with the globals
-	config.GlobalSettings[option] = nativeValue
-	config.ModifiedSettings[option] = true
+	if !reflect.DeepEqual(config.GlobalSettings[option], nativeValue) {
```

Comment:
Maybe useful, but still not fully preventing unneeded rewriting of `settings.json`: we still rewrite it (even multiple times) when `SetGlobalOptionNative()` is called from `reloadRuntime()`, i.e. when the setting has changed, yet we don't need to write it, since this new setting is already in `settings.json`, by definition (since we have just read it from `settings.json`)?

Reply:
After the change the question is, if this would be useful for `OverwriteSettings()` too?